pyccel/ast/cuda.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+CUDA Module
+This module provides a collection of classes and utilities for CUDA programming.
+"""
+from pyccel.ast.core import FunctionCall
+
+__all__ = (
+    'KernelCall',
+)
+
+class KernelCall(FunctionCall):
+    """
+    Represents a kernel function call in the code.
+    
+     The class serves as a representation of a kernel
+     function call within the codebase.
+
+    Parameters
+    ----------
+    func : FunctionDef
+        The definition of the function being called.
+
+    args : List[LiteralInteger]
+        List of LiteralInteger passed to the function.
```

Comment:
What do these literal integers represent? The arguments of the kernel function? So you allow:
```python
f[1,2](1,2)
f[num_blocks,tp_block](1,2)
```
but not:
```python
@kernel
def f(a_arg : int, b_arg : int):
    pass

@kernel
def g(arg : float):
    pass

a = 1
b = 2
f[1,2](a,b)
f[num_blocks,tp_block](1,2)
g[1,2](3.5)
```
?
Is this right? Please explain why your implementation needs to be so restrictive

Reply:
E.g. according to your docstring, your code does not handle this test code:
https://github.com/pyccel/pyccel-cuda/blob/f32042fb5018e3cb20585dbb3009fe866edbe383/tests/pyccel/scripts/kernel/hello_kernel.py#L14

Reply:
[162c5e8](https://github.com/pyccel/pyccel-cuda/pull/42/commits/162c5e81e51779dd936a0566cecbdc03fe5efdc7)